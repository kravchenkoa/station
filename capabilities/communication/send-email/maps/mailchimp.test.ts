import { MappedHTTPError, Provider } from '@superfaceai/one-sdk';

import {
  CommunicationSendEmailProfile,
  SuperfaceClient,
} from '../../../../superface/sdk';

describe('communication/send-email/mailchimp', () => {
  let client: InstanceType<typeof SuperfaceClient>;
  let profile: CommunicationSendEmailProfile;
  let provider: Provider;

  beforeEach(async () => {
    client = new SuperfaceClient();
    profile = await client.getProfile('communication/send-email');
    provider = await client.getProvider('mailchimp');
  });

  it('should find provider', async () => {
    expect(provider).not.toBeUndefined();
  });

  describe('when all inputs are correct', () => {
    it('should return messagaId as result', async () => {
      const result = await profile.useCases.SendEmail.perform(
        {
          from: process.env.MAILCHIMP_COMMUNICATION_SENDEMAIL_FROM,
          to: process.env.COMMUNICATION_SENDEMAIL_TO,
          subject: 'Station test',
          text: 'Station test',
        },
        { provider }
      );

      expect(typeof result.unwrap().messageId).toBe('string');
    });
  });

  describe('when inputs are invalid', () => {
    it('should throw error on unwrap', async () => {
      const result = await profile.useCases.SendEmail.perform(
        {
          to: 'invalidemail',
          from: 'invalidemail',
          subject: '',
          text: '',
        },
        { provider }
      );

      try {
        result.unwrap();

        // To ensure test continues in catch branch even if unwrap doesn't throw
        throw new Error('Unwrap should throw');
      } catch (error) {
        expect(error).toBeInstanceOf(MappedHTTPError);
        expect(error.properties.title).toBe('Invalid inputs');
      }
    });
  });
});
