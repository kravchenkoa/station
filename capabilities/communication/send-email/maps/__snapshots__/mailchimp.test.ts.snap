// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`communication/send-email/mailchimp SendEmail when all inputs are correct should return messageId as result 1`] = `
Object {
  "value": Object {
    "messageId": "67c75f4cc6814732af7b4a53446a74a8",
  },
}
`;

exports[`communication/send-email/mailchimp SendEmail when inputs are invalid should throw an exception 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[3].responseHandlers[1].statements[1]
Original Map Location: Line 51, column 7",
}
`;
